### PR TITLE
First leader vote

### PR DESCRIPTION
**Pull request by colorlace**
- Voting for a leader when there is no leader makes them the leader automatically (update to VoteHandler)
- When there is no leader, you can't vote for yourself (update to services/available_targets)

### CODE REVIEW

File: spec/services/available_targets_spec.rb [hunk 1]
```diff
@@ -12,22 +12,33 @@
   subject { AvailableTargets.new(action) }
 
   context "VoteLeader" do
-
```
Comment (efatsi) [inline spec/services/available_targets_spec.rb:15]: where's the :chefs_kiss: emoji?

**Review by efatsi — APPROVED** [review]
nice nice. should we add to the future todo list to make the first vote action a distinct "AppointLeader" action?

File: spec/services/available_targets_spec.rb (hunk 1 above)
Comment (colorlace) [reply spec/services/available_targets_spec.rb:15]: I don't see why we would need to make an AppointLeader action, but maybe we will for some reason.

**Comment by efatsi** [comment]
Merging this into `main`

**Pull request closed without merging**